### ISSUE TITLE
Modify naming structure

### ISSUE THREAD

**Issue by jenspots** [jenspots/libwheel#4]
Functions are currently named as follows

```c
function: <data_structure>_<name>_<type>
```

However, typically generics tend to switch up the order

```c
function: <data_structure>_<type>_<name>
```

This is more like a language such as Java and C++, where you have class generics such as `List<Integer>`. A notable exception is Rust, but only sometimes. For example, the function `parse::<int>`, which really uses the type more like a compile-time function parameter than anything else.

The `LIBWHEEL_PREFIX` macro would need to be modified. Preferable using some sort of internal `LIBWHEEL_DATASTRUCTURE_NAME` preprocessor definition.

**Comment by jenspots** [comment]
The macro prefix `LIBWHEEL_` is quite a mouthful, so it'd be better to switch to `LW_` or simply `W_`.

Also, it'd be better to simply get rid of the `T` alias of `LIBWHEEL_TYPE` and just use something like `LW_TYPE` directly.

**Comment by jenspots** [comment]
This has been resolved in commit 3b2d2d75e612fe625bf798b533458a4c9ce23f06.
